storybook/stories/Examples/Tooltip.stories.tsx

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable no-shadow */
+import React from 'react';
+import { pageData } from '../data';
+import { Bar, ComposedChart, Line, ResponsiveContainer, Tooltip } from '../../../src';
+import { DefaultTooltipContent } from '../../../src/component/DefaultTooltipContent';
+
+export default {
+  component: Tooltip,
+};
+
+export const LockedByClick = {
+  render: (_args: Record<string, any>) => {
+    const [isLocked, setIsLocked] = React.useState(false);
+    // The TooltipData contains the payload, the label and the x position of the tooltip.
+    // Their update is interrupted by the click event, so we need to store them in a state.
+    const [tooltipData, setTooltipData] = React.useState<{ payload?: unknown[]; label?: string; x?: number }>({});
+
+    // A custom Tooltip that updates the payload of the tooltip if the chart is locked, and either way always renders using the normal Tooltip.
+    const CustomTooltip = (props: any) => {
```

Comment:
we know at least somewhat what the type of CustomTooltip is? Is it possible to partially type this?